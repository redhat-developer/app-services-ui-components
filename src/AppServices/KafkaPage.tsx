import {
  Button,
  ButtonVariant,
  Card,
  CardBody,
  CardFooter,
  CardHeader,
  CardHeaderMain,
  CardTitle,
  Grid,
  Title,
} from "@patternfly/react-core";
import type { FunctionComponent } from "react";
import { useTranslation } from "react-i18next";
import { AppSpeedKeyArt } from "../images";
import { MarketingPageHero, MarketingPageSection } from "./components";
import { ExternalLinkAltIcon } from "@patternfly/react-icons";

export const KafkaPage: FunctionComponent = () => {
  const { t } = useTranslation();
  return (
    <>
      <MarketingPageHero
        title={t("kafkaoverview:heroTitle")}
        tagLine={t("kafkaoverview:heroTagline")}
        description={t("kafkaoverview:heroDescription")}
        heroImage={AppSpeedKeyArt}
        heroImageSize={478}
        heroImageCanRepeat={false}
        heroImagePositionY={-99}
      />
      <MarketingPageSection>
        <Grid hasGutter lg={6}>
          <Card
            ouiaId="card-overview-purchase-now"
            aria-label={t("kafkaoverview:purchaseCardTitle")}
          >
            <CardHeader>
              <CardHeaderMain>
                <CardTitle>
                  <Title headingLevel="h3">
                    {t("kafkaoverview:purchaseCardTitle")}
                  </Title>
                </CardTitle>
              </CardHeaderMain>
            </CardHeader>
            <CardBody>{t("kafkaoverview:purchaseCardMainText")}</CardBody>
            <CardFooter>
              <Button
                data-testid="cardPurchase-buttonCTA"
                variant={ButtonVariant.secondary}
                component="a"
                href="https://marketplace.redhat.com/en-us/products/red-hat-openshift-streams-for-apache-kafka"
                target="_blank"
                ouiaId="link-purchase-now"
              >
                {t("kafkaoverview:purchaseCardCallToActionButton")}
                <ExternalLinkAltIcon className="pf-u-ml-md" />
              </Button>
            </CardFooter>
          </Card>
          <Card
            ouiaId="card-overview-contact-sales"
            aria-label={t("kafkaoverview:contactSalesCardTitle")}
          >
            <CardHeader>
              <CardTitle>
                <Title headingLevel="h3">
                  {t("kafkaoverview:contactSalesCardTitle")}
                </Title>
              </CardTitle>
            </CardHeader>
            <CardBody>{t("kafkaoverview:contactSalesCardMainText")}</CardBody>
            <CardFooter>
              <Button
                data-testid="cardContactSales-buttonCTA"
                variant={ButtonVariant.secondary}
                component="a"
                href="https://www.redhat.com/en/technologies/cloud-computing/openshift/openshift-streams-for-apache-kafka#contact-sales"
                target="_blank"
                ouiaId="link-contact-sales"
              >
                {t("kafkaoverview:contactSalesCardCallToActionButton")}
                <ExternalLinkAltIcon className="pf-u-ml-md" />
              </Button>
            </CardFooter>
          </Card>
        </Grid>
      </MarketingPageSection>
    </>
  );
};
