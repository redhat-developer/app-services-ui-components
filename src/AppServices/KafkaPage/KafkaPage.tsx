import {
  Button,
  ButtonVariant,
  Card,
  CardBody,
  CardFooter,
  CardHeader,
  CardHeaderMain,
  CardTitle,
  Grid,
  Title,
} from "@patternfly/react-core";
import { ExternalLinkAltIcon } from "@patternfly/react-icons/dist/js/icons/external-link-alt-icon";
import { FunctionComponent } from "react";
import { useTranslation } from "react-i18next";
import { MarketingPageHero, MarketingPageSection } from "../components";
import { AppSpeedKeyArt } from "../../images";

export const KafkaPage: FunctionComponent = () => {
  const { t } = useTranslation();
  return (
    <>
      <MarketingPageHero
        title={t("kafkaoverview:heroTitle")}
        tagLine={t("kafkaoverview:heroTagline")}
        description={t("kafkaoverview:heroDescription")}
        heroImage={AppSpeedKeyArt}
        heroImageSize={478}
        heroImageCanRepeat={false}
        heroImagePositionY={-99}
      />
      <MarketingPageSection>
        <Grid hasGutter lg={6}>
          <Card>
            <CardHeader>
              <CardHeaderMain>
                <CardTitle>
                  <Title headingLevel="h3">
                    {t("kafkaoverview:purchaseCardTitle")}
                  </Title>
                </CardTitle>
              </CardHeaderMain>
            </CardHeader>
            <CardBody>{t("kafkaoverview:purchaseCardMainText")}</CardBody>
            <CardFooter>
              <Button
                data-testid="cardPurchase-buttonCTA"
                variant={ButtonVariant.secondary}
                component="a"
                href="https://marketplace.redhat.com/en-us/products/red-hat-openshift-streams-for-apache-kafka"
                target="_blank"
              >
                {t("kafkaoverview:purchaseCardCallToActionButton")}
                <ExternalLinkAltIcon className="pf-u-ml-md" />
              </Button>
            </CardFooter>
          </Card>
          <Card>
            <CardHeader>
              <CardTitle>
                <Title headingLevel="h3">
                  {t("kafkaoverview:contactSalesCardTitle")}
                </Title>
              </CardTitle>
            </CardHeader>
            <CardBody>{t("kafkaoverview:contactSalesCardMainText")}</CardBody>
            <CardFooter>
              <Button
                data-testid="cardContactSales-buttonCTA"
                variant={ButtonVariant.secondary}
                component="a"
                href="https://www.redhat.com/en/technologies/cloud-computing/openshift/openshift-streams-for-apache-kafka#contact-sales"
                target="_blank"
              >
                {t("kafkaoverview:contactSalesCardCallToActionButton")}
                <ExternalLinkAltIcon className="pf-u-ml-md" />
              </Button>
            </CardFooter>
          </Card>
        </Grid>
      </MarketingPageSection>
    </>
  );
};
