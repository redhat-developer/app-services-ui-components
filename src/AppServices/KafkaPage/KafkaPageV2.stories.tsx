import { ComponentMeta, ComponentStory } from "@storybook/react";
import { KafkaPageV2 } from "./KafkaPageV2";

export default {
  component: KafkaPageV2,
  args: {},
} as ComponentMeta<typeof KafkaPageV2>;

const Template: ComponentStory<typeof KafkaPageV2> = (args) => (
  <KafkaPageV2 {...args} />
);

export const Example = Template.bind({});
