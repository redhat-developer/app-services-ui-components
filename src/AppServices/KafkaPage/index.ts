export * from "./KafkaPage";
export * from "./KafkaPageV2";
