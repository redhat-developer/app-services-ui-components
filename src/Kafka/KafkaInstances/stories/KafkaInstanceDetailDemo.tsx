import {
  Breadcrumb,
  BreadcrumbItem,
  Button,
  Level,
  LevelItem,
  PageSection,
  Skeleton,
  Tab,
  TabContent,
  Tabs,
  TabsComponent,
  Title,
} from "@patternfly/react-core";
import type { ComponentType } from "react";
import { lazy, Suspense, useRef } from "react";
import {
  Link,
  Redirect,
  Route,
  Switch,
  useHistory,
  useParams,
  useRouteMatch,
} from "react-router-dom";
import { Loading } from "../../../shared";
import { useKafkaInstanceDrawer } from "../KafkaInstanceDrawer";
import { useInstance } from "./support";

const LazyDashboard = lazy(
  () =>
    new Promise<{ default: ComponentType<any> }>((resolve) =>
      setTimeout(() => resolve(import("./SampleDashboard")), 1500)
    )
);

const LazyTopics = lazy(
  () =>
    new Promise<{ default: ComponentType<any> }>((resolve) =>
      setTimeout(() => resolve(import("./SampleTopics")), 1500)
    )
);

export const KafkaInstanceDetailDemo = () => {
  const { id, tab = "dashboard" } = useParams<{
    id: string;
    tab: string | undefined;
  }>();
  const history = useHistory();
  const match = useRouteMatch({ path: "/streams/:id" });
  const { instance } = useInstance(id);
  const { openDrawer } = useKafkaInstanceDrawer();
  const dashboardRef = useRef<HTMLDivElement>(null);
  const topicsRef = useRef<HTMLDivElement>(null);
  return (
    <>
      <PageSection variant={"light"} type={"breadcrumb"}>
        <Breadcrumb>
          <BreadcrumbItem
            render={() => <Link to={"/streams"}>Kafka instances</Link>}
          />
          <BreadcrumbItem isActive>
            {instance?.name || <Skeleton width={"150px"} />}
          </BreadcrumbItem>
        </Breadcrumb>
      </PageSection>
      <PageSection variant={"light"}>
        <Level>
          <LevelItem>
            <Title headingLevel={"h1"}>
              {instance?.name || <Skeleton width={"150px"} />}
            </Title>
          </LevelItem>
          <LevelItem>
            {instance && (
              <Button onClick={() => openDrawer(instance)}>Open drawer</Button>
            )}
          </LevelItem>
        </Level>
      </PageSection>
      <PageSection variant={"light"} type={"tabs"}>
        <Tabs
          activeKey={tab}
          onSelect={(_, key) => {
            history.push({
              pathname: `${match?.url || ""}/${key}`,
            });
          }}
          component={TabsComponent.nav}
          aria-label="Tabs in the toggled separate content example"
        >
          <Tab
            eventKey={"dashboard"}
            title="Dashboard"
            tabContentId="dashboard-tab"
          />
          <Tab eventKey={"topics"} title="Topics" tabContentId="topics-tab" />
        </Tabs>
      </PageSection>
      <PageSection isFilled={true}>
        <Suspense fallback={<Loading />}>
          <Switch>
            <Route path={`${match?.path || ""}/dashboard`}>
              <TabContent
                eventKey={"dashboard"}
                id="dashboard-tab"
                ref={dashboardRef}
                aria-label="TODO"
              >
                <LazyDashboard />
              </TabContent>
            </Route>
            <Route path={`${match?.path || ""}/topics`}>
              <TabContent
                eventKey={"topics"}
                id="topics-tab"
                ref={topicsRef}
                aria-label="TODO"
              >
                <LazyTopics />
              </TabContent>
            </Route>
            <Redirect
              path={match?.path}
              exact={true}
              to={`${match?.path || ""}/dashboard`}
            />
          </Switch>
        </Suspense>
      </PageSection>
    </>
  );
};

export default KafkaInstanceDetailDemo;
