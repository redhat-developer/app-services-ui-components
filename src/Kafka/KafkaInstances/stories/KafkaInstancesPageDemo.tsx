import { KafkaInstancesPage } from "../KafkaInstancesPage";
import { useInstances } from "./support";

export const KafkaInstancesPageDemo = () => {
  const { instances } = useInstances();

  return (
    <KafkaInstancesPage
      instances={instances}
      itemCount={instances?.length}
      page={1}
      getUrlForInstance={(row) => `/streams/${row.id}`}
      onChangeOwner={() => {}}
      onClickConnectionTabLink={() => {}}
      onClickSupportLink={() => {}}
      onConnection={() => {}}
      onCreate={() => {}}
      onDelete={() => {}}
      onDetails={() => {}}
      onPageChange={() => {}}
      onQuickstartGuide={() => {}}
    />
  );
};

export default KafkaInstancesPageDemo;
