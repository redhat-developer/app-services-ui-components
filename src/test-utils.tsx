import type { RenderOptions, RenderResult } from "@testing-library/react";
import { act, render, waitFor } from "@testing-library/react";
import type { FunctionComponent, ReactElement } from "react";
import { Suspense } from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { I18nProvider } from "./I18n";
import { AlertProvider } from "../src/shared/Alerts";

const suspenseTestId = "i18n-suspense";

const AllTheProviders: FunctionComponent = ({ children }) => {
  return (
    <Router>
      <AlertProvider>
        <I18nProvider
          lng={"en"}
          resources={{
            en: {
              common: () => import("../locales/en/common.json"),
              "create-kafka-instance": () =>
                import("../locales/en/create-kafka-instance.json"),
              kafka: () => import("../locales/en/kafka.json"),
              metrics: () => import("../locales/en/metrics.json"),
              overview: () => import("../locales/en/overview.json"),
              datascienceoverview: () =>
                import("../locales/en/datascienceoverview.json"),
              kafkaoverview: () => import("../locales/en/kafkaoverview.json"),
              topic: () => import("../locales/en/topic.json"),
              apimgmtoverview: () =>
                import("../locales/en/apimgmtoverview.json"),
              "manage-kafka-permissions": () =>
                import("../locales/en/manage-kafka-permissions.json"),
              "create-topic": () => import("../locales/en/create-topic.json"),
              "message-browser": () =>
                import("../locales/en/message-browser.json"),
              "overview-v2": () => import("../locales/en/overview-v2.json"),
              "overview-v3": () => import("../locales/en/overview-v3.json"),
              "overview-v4": () => import("../locales/en/overview-v4.json"),
              "kafkaoverview-v2": () =>
                import("../locales/en/kafkaoverview-v2.json"),
              "kafkaoverview-v3": () =>
                import("../locales/en/kafkaoverview-v3.json"),
              "connection-tab": () =>
                import("../locales/en/connection-tab.json"),
              "service-registry": () =>
                import("../locales/en/service-registry.json"),
              "connection-tab-p1": () =>
                import("../locales/en/connection-tab-p1.json"),
            },
            it: {
              common: () => Promise.resolve({ delete: "Elimina" }),
            },
          }}
          debug={false}
        >
          <Suspense fallback={<div data-testid={suspenseTestId}>loading</div>}>
            {children}
          </Suspense>
        </I18nProvider>
      </AlertProvider>
    </Router>
  );
};

const AllTheProvidersWithRoot: FunctionComponent = ({ children }) => (
  <AllTheProviders>
    <div id={"root"}>{children}</div>
  </AllTheProviders>
);
const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

const renderDialog = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProvidersWithRoot, ...options });

async function waitForI18n(r: RenderResult) {
  await waitFor(() => {
    expect(r.queryByTestId(suspenseTestId)).not.toBeInTheDocument();
  });
}

async function waitForPopper() {
  await act(async () => {
    /* let popper do its updates */
  });
}

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderDialog, waitForI18n, waitForPopper };
