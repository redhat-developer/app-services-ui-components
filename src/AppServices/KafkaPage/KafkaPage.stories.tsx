import { ComponentMeta, ComponentStory } from "@storybook/react";
import { KafkaPage } from "./KafkaPage";

export default {
  component: KafkaPage,
  args: {},
} as ComponentMeta<typeof KafkaPage>;

const Template: ComponentStory<typeof KafkaPage> = (args) => (
  <KafkaPage {...args} />
);

export const Example = Template.bind({});
