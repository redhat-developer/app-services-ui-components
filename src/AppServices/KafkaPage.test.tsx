import { composeStories } from "@storybook/testing-react";
import { render, waitForI18n } from "../test-utils";
import * as stories from "./KafkaPage.stories";

const { KafkaPage } = composeStories(stories);

describe("KafkaPage", () => {
  it("renders", async () => {
    const comp = render(<KafkaPage />);
    await waitForI18n(comp);
    expect(comp.baseElement).toMatchSnapshot();
  });

  it("should persist ouia ids", async () => {
    const comp = render(<KafkaPage />);
    await waitForI18n(comp);
    expect(comp.getByRole("article", { name: "Purchase now" })).toHaveAttribute(
      "data-ouia-component-id",
      "card-overview-purchase-now"
    );
    expect(comp.getByRole("link", { name: "Purchase now" })).toHaveAttribute(
      "data-ouia-component-id",
      "link-purchase-now"
    );

    expect(
      comp.getByRole("article", { name: "Contact sales" })
    ).toHaveAttribute("data-ouia-component-id", "card-overview-contact-sales");
    expect(comp.getByRole("link", { name: "Contact sales" })).toHaveAttribute(
      "data-ouia-component-id",
      "link-contact-sales"
    );
  });
});
