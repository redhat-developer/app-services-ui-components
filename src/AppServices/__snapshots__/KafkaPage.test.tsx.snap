// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KafkaPage renders 1`] = `
<body>
  <div>
    <section
      class="pf-c-page__main-section pf-m-light appsrv-marketing--banner pf-u-background-color-100"
      style="--appsrv-marketing--banner--before--BackgroundImage: url(dummyimage); --appsrv-marketing--banner--before--BackgroundSize: 478px; --appsrv-marketing--banner--before--BackgroundRepeat: no-repeat; --appsrv-marketing--banner--before--BackgroundPositionY: -99px;"
    >
      <div
        class="pf-l-stack pf-m-gutter"
      >
        <h1
          class="pf-c-title pf-m-2xl"
          data-ouia-component-id="OUIA-Generated-Title-1"
          data-ouia-component-type="PF4/Title"
          data-ouia-safe="true"
        >
          Get started with Red Hat OpenShift Streams for Apache Kafka
        </h1>
        <p
          class="appsrv-marketing--banner__tagline pf-u-color-200"
          data-ouia-component-id="OUIA-Generated-Text-1"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          Fully hosted and managed service for stream-based applications
        </p>
        <p
          class=""
          data-ouia-component-id="OUIA-Generated-Text-2"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          Red Hat OpenShift Streams for Apache Kafka is a managed cloud service that provides a streamlined developer experience for building, deploying, and scaling real-time applications in hybrid-cloud environments. The combination of seamless operations across distributed microservices, large data transfer volumes, and managed operations allows teams to focus on core competencies, accelerate application velocity and reduce operational cost.
        </p>
      </div>
    </section>
    <section
      class="pf-c-page__main-section pf-m-limit-width appsrv-marketing--page-section--marketing"
    >
      <div
        class="pf-c-page__main-body"
      >
        <div
          class="pf-l-grid pf-m-all-6-col-on-lg pf-m-gutter"
        >
          <article
            aria-label="Purchase now"
            class="pf-c-card"
            data-ouia-component-id="card-overview-purchase-now"
            data-ouia-component-type="PF4/Card"
            data-ouia-safe="true"
            id=""
          >
            <div
              class="pf-c-card__header"
            >
              <div
                class=""
              >
                <div
                  class="pf-c-card__title"
                >
                  <h3
                    class="pf-c-title pf-m-lg"
                    data-ouia-component-id="OUIA-Generated-Title-2"
                    data-ouia-component-type="PF4/Title"
                    data-ouia-safe="true"
                  >
                    Purchase now
                  </h3>
                </div>
              </div>
            </div>
            <div
              class="pf-c-card__body"
            >
              Use Red Hat Marketplace to purchase a pay-as-you-go subscription for up to 25 streaming units. Purchasing through Red Hat Marketplace is only available in the United States and Canada.
            </div>
            <div
              class="pf-c-card__footer"
            >
              <a
                aria-disabled="false"
                class="pf-c-button pf-m-secondary"
                data-ouia-component-id="link-purchase-now"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                data-testid="cardPurchase-buttonCTA"
                href="https://marketplace.redhat.com/en-us/products/red-hat-openshift-streams-for-apache-kafka"
                target="_blank"
              >
                Purchase now
                <svg
                  aria-hidden="true"
                  class="pf-u-ml-md"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                  />
                </svg>
              </a>
            </div>
          </article>
          <article
            aria-label="Contact sales"
            class="pf-c-card"
            data-ouia-component-id="card-overview-contact-sales"
            data-ouia-component-type="PF4/Card"
            data-ouia-safe="true"
            id=""
          >
            <div
              class="pf-c-card__header"
            >
              <div
                class="pf-c-card__title"
              >
                <h3
                  class="pf-c-title pf-m-lg"
                  data-ouia-component-id="OUIA-Generated-Title-3"
                  data-ouia-component-type="PF4/Title"
                  data-ouia-safe="true"
                >
                  Contact sales
                </h3>
              </div>
            </div>
            <div
              class="pf-c-card__body"
            >
              Contact sales to get a pre-paid subscription or modify limits of your current subscription. Pre-paid subscriptions are available to customers in any location, including locations outside the United States and Canada.
            </div>
            <div
              class="pf-c-card__footer"
            >
              <a
                aria-disabled="false"
                class="pf-c-button pf-m-secondary"
                data-ouia-component-id="link-contact-sales"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                data-testid="cardContactSales-buttonCTA"
                href="https://www.redhat.com/en/technologies/cloud-computing/openshift/openshift-streams-for-apache-kafka#contact-sales"
                target="_blank"
              >
                Contact sales
                <svg
                  aria-hidden="true"
                  class="pf-u-ml-md"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                  />
                </svg>
              </a>
            </div>
          </article>
        </div>
      </div>
    </section>
  </div>
  <div>
    <ul
      aria-atomic="false"
      aria-live="polite"
      class="pf-c-alert-group pf-m-toast"
    />
  </div>
</body>
`;
